### ISSUE TITLE
Closing non-existing workspace

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#182]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/182)
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Jan 7, 2018, 12:58

When running `bst workspace close --remove-dir core/gnome-keyring.bst`, the user is asked if they really want to remove the directory, because it can imply loss of work.

After answering "Yes I wanna delete that...", BuildStream errors out with:

```
ERROR: Workspace 'core/gnome-keyring.bst - 0' is currently not defined
``` 

This is backwards, the error message should come before ever prompting the user.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @knownexus]](https://gitlab.com/knownexus) on Jan 12, 2018, 11:47

i'll have a look at this and see what i can do

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @knownexus]](https://gitlab.com/knownexus) on Jan 12, 2018, 12:37

mentioned in merge request !227

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Feb 13, 2018, 05:19

closed via commit b5f43ece275638b5ba4bff817ba3379dd9ae936a

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Feb 13, 2018, 05:42

mentioned in commit b5f43ece275638b5ba4bff817ba3379dd9ae936a

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Mar 20, 2018, 14:03

mentioned in merge request !317

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Apr 16, 2018, 18:22

Commit 19e31adb00f19f0789a3c4622150ca8f6e9c8901 from !352 reintroduced this issue. Reopening.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Apr 16, 2018, 18:22

reopened

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 16, 2018, 18:40

That's odd, I thought we have regression tests for this case now.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Apr 16, 2018, 18:42

The issue only exists in interactive mode, which we don't test, as far as I can tell.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 17, 2018, 08:09

mentioned in commit 9cf8d35fce45b85de3535815501af92e175faa3b

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 17, 2018, 08:37

closed via merge request !408

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 17, 2018, 08:37

closed via commit 9cf8d35fce45b85de3535815501af92e175faa3b
